cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -320,4 +340,25 @@ private void popStash(@Nullable RevCommit stashID) throws GitAPIException {
     LOG.warn("Couldn't find Git stash with ID {} to drop. Ignoring missing stash.", stashID.getName());
   }
 
+  /**
+   * Does a <a href="https://git-scm.com/docs/git-reset"> hard reset </a> to the remote branch being tracked.
+   */
+  private String resetToCleanBranch() throws IOException, GitAPIException {
+    List<RemoteConfig> remoteConfigs = git.remoteList().call();
+    if (remoteConfigs.isEmpty()) {
+      throw new IllegalStateException("No remotes found for git repository.");
+    }
+    if (remoteConfigs.size() > 1) {
```

Comment:
nit: you can do `!= 1` and use 1 check instead of 2. Anyway this is something that should never fail. Still good to validate though

Reply:
Merged both the checks.